### DISCUSSION TITLE
What exactly is quarkus.oidc.client-id?

### DISCUSSION THREAD

**Discussion by gefloh** [quarkusio/quarkus#38039] [Q&A] [answered]
Hi,

can anyone explain to me what exactly the OIDC parameter `quarkus.oidc.client-id` means in Quarkus?

* Javadoc: "The client-id of the application. Each application has a client-id that is used to identify the application"
* Quarkus guide: "quarkus.oidc.client-id sets a client-ID that identifies the application."

Ok, but where is this actually used? In my application (which uses OIDC Bearer Auth) I can even remove this parameter completely and everything is still working fine. Or is this only relevant for the authorization code flow (which I am not using)?

Thanks,
Gerhard

[a comment by an automated account is not shown]

**Comment by gefloh** [comment, marked as answer]
Silly me, there is even a whole paragraph about it [here](https://quarkus.io/guides/security-oidc-bearer-token-authentication#how-to-use-client-id-property). 

Sometimes it's a bit difficult to keep track because there are so many guides in the OIDC area (which is of course much appreciated on the other side).

**Comment by sberyozkin** [comment]
Thanks @gefloh
